components/raftstore/src/store/peer.rs

```diff
@@ -1513,59 +1529,46 @@ where
-                            .unwrap()
-                            .last()
-                            .unwrap()
-                            .index,
-                        hs.get_commit()
-                    );
-                    let mut split_to_be_updated = true;
-                    let mut merge_to_be_updated = true;
-                    for entry in ready.committed_entries.as_ref().unwrap().iter().rev() {
-                        // We care about split/merge commands that are committed in the current term.
-                        if entry.term == self.term() && (split_to_be_updated || merge_to_be_updated)
-                        {
-                            let ctx = ProposalContext::from_bytes(&entry.context);
-                            if split_to_be_updated && ctx.contains(ProposalContext::SPLIT) {
-                                // We don't need to suspect its lease because peers of new region that
-                                // in other store do not start election before theirs election timeout
-                                // which is longer than the max leader lease.
-                                // It's safe to read local within its current lease, however, it's not
-                                // safe to renew its lease.
-                                self.last_committed_split_idx = entry.index;
-                                split_to_be_updated = false;
-                            } else if merge_to_be_updated
-                                && ctx.contains(ProposalContext::PREPARE_MERGE)
-                            {
-                                // We committed prepare merge, to prevent unsafe read index,
-                                // we must record its index.
-                                self.last_committed_prepare_merge_idx = entry.get_index();
-                                // After prepare_merge is committed, the leader can not know
-                                // when the target region merges majority of this region, also
-                                // it can not know when the target region writes new values.
-                                // To prevent unsafe local read, we suspect its leader lease.
-                                self.leader_lease.suspect(monotonic_raw_now());
-                                merge_to_be_updated = false;
-                            }
+                    // The admin cmds in `CmdEpochChecker` are proposed by the current leader so we can use
+                    // it to get the split/prepare-merge cmds which was committed just now.
+
+                    // BatchSplit and Split cmd are mutually exclusive because they both change epoch's version
+                    let last_split_idx = self
+                        .cmd_epoch_checker
+                        .last_cmd_index(AdminCmdType::BatchSplit)
+                        .or_else(|| self.cmd_epoch_checker.last_cmd_index(AdminCmdType::Split));
```

Comment:
I think it should be the maximum of them instead of choosing one.

Reply:
These two cmds are mutually exclusive because they both change epoch's version so only one of them will be proposed and the other one will be rejected by `CmdEpochChecker`.